### PR TITLE
Add setFocusable to CoachMarkBuilder

### PR DESCRIPTION
**Pull request by jianclv**
Provide a function to let user setFocusable from outside to meet different requirements.
Such as:
1. We want user can only do interaction with this coach mark, outside coach mark need become un-touchable.
2. In talkback mode, we want restrict focus inner this coach mark area, other than can go out and below of the coach mark.

Here are some pictures to show the behavior change:
cornedbeef1
cornedbeef2

Pic 1 is the original screenshot before apply this change, the following pics are after apply this change.
Pic 2 is in non EbT mode, same UI as before, if we want quit the app by click hardware back button, it may need several click as each click dismiss one coach mark.
Pic 3 shows that the last coach mark get focus, in this time, focus can't move outside of that coach mark.
Pic 4 shows after perform one back button click, the last coach mark disappear.
Pic 5 shows after perform one more back button click, the second from last coach mark disappear.

### CODE REVIEW

**Comment by duffin22** [comment]
Have you checked that this works as expected with all types of screenshots? If so, it would be good to see some screenshots.

Also, is this something that you would be able to add to the automated tests?

**Comment by JPumphrey** [comment]
With EbT off, would areas outside the coachmark be touchable?

**Comment by jianclv** [comment]
With Explore by Touch mode off, same as in EbT mode, the areas outside is not touchable as well.

**Comment by jianclv** [comment]
Screenshots added, this setFocusable don't change any UI appearance. Added a simple test case. After this PR, the UI and behavior should exactly same as before if developer don't call setFocusable. If developer called it, it's the developer's responsibility to know the behavior change. I personally think setFocusable is quite useful in some situation and it's necessary to expose it to user. Developer can do some strategy to call it, such as call setFocusable(isEbTMode) to make coach mark focusable only in EbT mode.

**Comment by duffin22** [comment]
> With Explore by Touch mode off, same as in EbT mode, the areas outside is not touchable as well.

This doesn't sound desirable. Does touching the area outside of the coach mark still dismiss a bubble coach mark if ebt is off? It would be good if this `setFocusable` behaves nicely both with ebt on and off so we don't make life harder for developers.

**Comment by jianclv** [comment]
My bad, I may say something wrong before, With Explore by Touch mode off, the areas outside is *touchable*, but it will dismiss the coach mark if user touch outside the coach mark.
I just tested, for LayeredCoachMark, PunchHoleCoachMark and BubbleCoachMark it works (touch outside will dismiss the coach mark), however, HighlightCoachMark not works in this case as it called popup.setTouchable(false). I'll update comment for the code.

**Comment by jianclv** [comment]
@duffin22 Do you think add a function "setFocusableIfEbTMode()" helpful? Seems it will make life easier for developers. The isTouchExplorationEnabled() function starts from API 14 but our coach mark library support from API 12, we still can't do it for API 12 and 13.

**Comment by duffin22** [comment]
From your previous comment, seems like the behaviour would actually be okay without ebt? If the current behaviour without ebt is as before, this one method should suffice

File: cornedbeef/src/main/java/com/swiftkey/cornedbeef/CoachMark.java [hunk 1]
```diff
@@ -227,10 +227,15 @@ public boolean isShowing() {
      * Exposes the {@link PopupWindow#setFocusable(boolean)} method of {@link CoachMark#mPopup}
      *
      * Set whether the coach mark is focusable or not
-     * If we set coach mark focusable, 3 behaviors will change:
+     * If we set coach mark focusable, behaviors will change.
+     * For Explore-by-Touch mode
      *   1. Outside of this coach mark become un-touchable
      *   2. Coach mark can dismiss by pressing hardware back button
-     *   3. In Explore-by-Touch mode, focus will be restricted inner this coach mark area
+     *   3. Focus will be restricted inner this coach mark area
```
[the remark this answers is not in the document]
Comment (jianclv) [reply cornedbeef/src/main/java/com/swiftkey/cornedbeef/CoachMark.java]: What I mean is when user navigate focus through elements, only the elements inside the coach mark can be navigated.

File: cornedbeef/src/main/java/com/swiftkey/cornedbeef/CoachMark.java [hunk 2]
```diff
@@ -227,10 +227,15 @@ public boolean isShowing() {
...
+     * For Non Explore-by-Touch mode
+     *   1. Touch outside will dismiss the coach mark.
```
[the remark this answers is not in the document]
Comment (jianclv) [reply cornedbeef/src/main/java/com/swiftkey/cornedbeef/CoachMark.java]: For HighlightCoachMarks , nothing happen when touch outside.

File: cornedbeef/src/main/java/com/swiftkey/cornedbeef/CoachMark.java [hunk 3]
```diff
@@ -222,6 +222,33 @@ public View getContentView() {
...
         return mPopup.isShowing();
     }
+
+    /**
+     * Exposes the {@link PopupWindow#setFocusable(boolean)} method of {@link CoachMark#mPopup}
+     *
+     * Set whether the coach mark is focusable or not
+     * If we set the coach mark focusable, the behaviour changes as follows:
+     * For Explore-by-Touch mode
+     *   1. Outside of this coach mark become un-touchable
```
Comment (JPumphrey) [inline cornedbeef/src/main/java/com/swiftkey/cornedbeef/CoachMark.java]: `becomes`

**Pull request merged**